projects/laji/src/app/shared-modules/document-viewer/images/images.component.ts

```diff
@@ -50,13 +51,31 @@ export class ImagesComponent implements OnChanges {
           this.gatheringImages = this.gatheringImages.concat(gathering.media);
         }
         if (gathering.units) {
-          gathering.units.filter(unit => unit.unitId === this.highlight).map(
+          if (!this.identifying) {
+            gathering.units.filter(unit => unit.unitId === this.highlight).map(
               item => {
                 if (item.media) {
                   this.unitImages = this.unitImages.concat(item.media);
                 }
               }
-          );
+            );
+          } else {
+            let imgUnitId;
+            gathering.units.forEach(element => {
+              (element.media || []).forEach(image => {
+                if (image.fullURL === this.highlight) {
+                  imgUnitId = element.unitId;
+                }
+              });
+            });
+            gathering.units.filter(unit => unit.unitId === imgUnitId).map(
```

Comment:
This repeats almost the same logic twice. Instead you could do this kind of logic:
(highlight can now be unit id or image fullURL)
find the unit that has the highlight id or image that has the highlighted full url
and then add that units media to this.unitImages list
something like this:
``` 
if (!gathering.units) {
  return;
}
const unit = gathering.units.find(u => u.id === this.highlight || (u.media || []).some(media => media.fullURL === this.highlight));
if (unit?.media) {
  this.unitImages.concat(unit.media);
}